### PR TITLE
perf: add Count() query path for sidebar artist badge (#852)

### LINKED ISSUE #852: perf: add Count() service path for sidebar artist badge endpoint
## Problem

The `/api/v1/artists/badge` endpoint (used by the sidebar badge, polled every 120s) calls `artistService.List` with an empty filter, which hydrates full artist records including metadata and images. Only the `total` count is used.

## Proposed fix

Add a lightweight `Count()` path through the service and repository layers:

1. Add `Count(ctx context.Context, params CountParams) (int, error)` to `ArtistRepository` interface
2. Implement in `sqlite_artist.go` with a `SELECT COUNT(*)` query
3. Add `Service.Count()` that delegates to the repository
4. Update `handleArtistsBadge` in `handlers_artist.go` to call `Count()` instead of `List()`

## Context

Flagged by CodeRabbit on PR #847. Low priority since the poll interval is 120s, but avoids unnecessary DB and CPU work per request.

[mode: direct]
[model: sonnet]
[effort: low]

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary
- Add `Count(ctx, CountParams)` to the artist `Repository` interface and SQLite implementation using `SELECT COUNT(*)`
- Add `CountParams` type that mirrors filtering from `ListParams` but omits pagination/sorting
- Switch `handleArtistsBadge` from `List()` to `Count()`, eliminating row scanning and provider/image hydration
- Add 6 tests covering: empty DB, total count, library filter, search query, excluded filter, and consistency with List

## Test plan
- [x] Unit tests for Count with no artists
- [x] Unit tests for Count with various filters (library, search, excluded)
- [x] Verify Count returns same total as List for identical filters
- [ ] Verify sidebar badge shows correct count in browser
- [ ] Verify List() behavior unchanged

Closes #852

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Refactor**
  * Improved artist total retrieval to use a dedicated counting implementation for more efficient and accurate sidebar totals.

* **New Features**
  * Added a dedicated artist counting endpoint in the backend to support filtered total queries (no row data returned).

* **Tests**
  * Added comprehensive unit tests covering artist count behavior across filters, search, and library scoping.

### CODE REVIEW

File: internal/artist/count_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		Page:      1,
+		PageSize:  50,
+		LibraryID: "lib-1",
+	})
+	if err != nil {
+		t.Fatalf("List: %v", err)
+	}
+
+	if countTotal != listTotal {
+		t.Errorf("Count = %d, List total = %d; they should match", countTotal, listTotal)
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/artist/count_test.go]: Fixed in b163feb. Expanded into a table-driven test covering library_id, excluded, not_excluded, and library_id+excluded filter combinations.

**Comment by sydlexius** [comment]
@coderabbitai status
